src/stim/stabilizers/tableau.cc

```diff
@@ -379,6 +386,31 @@ bool Tableau::satisfies_invariants() const {
     return true;
 }
 
+bool Tableau::is_conjugation_by_pauli() {
+    simd_bit_table<MAX_BITWORD_WIDTH> scratch_matrix =
+        simd_bit_table<MAX_BITWORD_WIDTH>::identity(num_qubits);
+
+    if ( xs.xt != scratch_matrix || zs.zt != scratch_matrix ) {
```

Comment:
Instead of allocating, I would check if the backing array had a popcnt equal to num_qubits * 2 and that there was an X_k -> X_k and Z_k -> Z_k term for each string.

Reply:
done